### ISSUE TITLE
[Bug]: Adding integration outputs "Connection timeout" error

### ISSUE THREAD

**Issue by JohnyBeGood** [travisghansen/hass-opnsense#328]
### What happened?

Hey all,

I had integration working and upgraded the hardware. Removed integration from HACS and OPNsensee plugin. Installed everything by following the readme. All I'm getting now is "Connection timeout" every time I want to add it.
Any suggestions?

![image](https://github.com/user-attachments/assets/4a4e4210-da24-4e48-b2d3-b521edaf8248)
![image](https://github.com/user-attachments/assets/1b55ba38-57a3-450e-a6ed-7467916aeb83)

### hass-opnsense Version

0.3.9

### OPNsense Firmware

24.7.9_1

### Home Assistant Version

Core 2024.11.3

### Relevant logs

```shell
Home Assistant Core log:

Log details (ERROR)
This error originated from a custom integration.
Logger: custom_components.opnsense.config_flow
Source: custom_components/opnsense/config_flow.py:205
integration: opnsense (documentation, issues)
First occurred: 8:01:25 PM (4 occurrences)
Last logged: 8:12:15 PM

Timeout Error. TimeoutError:
```

### Additional Details

I can provide anything else if needed.

**Comment by Snuffy2** [comment]
Are you sure the address is correct for OPNsense?

**Comment by JohnyBeGood** [comment]
Thanks for the reply!

Yes. I copied/pasted it from the browser.

**Comment by alexdelprete** [comment]
From the HA bash shell (assuming you installed the SSH addon, etc.), please run this (copy&paste everything in the shell):

```bash
curl "https://192.168.1.1:443" -s -o /dev/null -w \
> "response_code: %{http_code}
> dns_time: %{time_namelookup}
> connect_time: %{time_connect}
> pretransfer_time: %{time_pretransfer}
> starttransfer_time: %{time_starttransfer}
> total_time: %{time_total}
> "
```

Output is this:

```bash
response_code: 200
dns_time: 0.029
connect_time: 0.046
pretransfer_time: 0.203
starttransfer_time: 0.212
total_time: 0.212
```

**Comment by JohnyBeGood** [comment]
Sure. My port is 4443. Here's the output:

`➜  ~ curl "https://192.168.1.1:4443" -s -o /dev/null -w \
"response_code: %{http_code}
dns_time: %{time_namelookup}
connect_time: %{time_connect}
pretransfer_time: %{time_pretransfer}
starttransfer_time: %{time_starttransfer}
total_time: %{time_total}
"
response_code: 000
dns_time: 0.000022
connect_time: 0.000238
pretransfer_time: 0.000000
starttransfer_time: 0.000000
total_time: 0.016756
➜  ~ `

**Comment by alexdelprete** [comment]
retry, I updated the commands. recheck the post.

**Comment by JohnyBeGood** [comment]
Sure. Here we go:

`
➜  ~ curl "https://192.168.1.1:4443" -fsko /dev/null -w \
"\n\
response_code: %{http_code}\n\
dns_time: %{time_namelookup}\n\
connect_time: %{time_connect}\n\
pretransfer_time: %{time_pretransfer}\n\
starttransfer_time: %{time_starttransfer}\n\
total_time: %{time_total}\n"

response_code: 200
dns_time: 0.000024
connect_time: 0.000313
pretransfer_time: 0.017775
starttransfer_time: 0.048148
total_time: 0.048240
➜  ~ 

`

**Comment by alexdelprete** [comment]
So it's not a network issue, response code 200 means the port is reachable from HA.

We need debug logs for the config flow to understand why the connect is timing out.

**Comment by Snuffy2** [comment]
I'm really not sure whats going on.

Please [enable debug logging](https://github.com/travisghansen/hass-opnsense/blob/main/CONTRIBUTING.md#enable-debug-logging-in-home-assistant), upgrade hass-opnsense to v0.3.12, and restart HA.

Then try to install the integration again. If it wasn't working with v0.3.11, I doubt it will work with v0.3.12. So assuming it still doesn't work, please post the debug logs here from the unsuccessful install.

**Comment by JohnyBeGood** [comment]
Solved! Thank you both for your time!

Since it was suggested to upgrade to hass-opnsense v0.3.12, when I went to the HACS I
 noticed that I was on v0.3.9 which is 2 months old (no wonder it didn't work). I went ahead and selected to re-download and install a specific version which at this time is v0.3.12. After the installation, I rebooted and started adding integration and entering the same IP:PORT and API key and secret and voila, it connected without time out.
I'm always up-to-date when it comes to HA and add-ons. I never got a notification there's a new version of hass-opnsense. 

Hope it helps someone!

Old version
![old v](https://github.com/user-attachments/assets/c168885e-5c76-46ea-ad69-6f52895e0cb6)

New version
![new v](https://github.com/user-attachments/assets/ac2341fb-f0b9-44a4-9947-b791bad425d6)

Success!
![success](https://github.com/user-attachments/assets/9e0e4e2e-1ba8-4ffd-914a-b8aa35aaf19f)
